src/Sylius/Bundle/CartBundle/Controller/CartItemController.php

```diff
@@ -137,6 +140,32 @@ public function removeAction(Request $request)
     }
 
     /**
+     * @param Request $request
+     * @param CartInterface $cart
+     */
+    protected function saveCartInSession(Request $request, CartInterface $cart)
+    {
+        $session = $request->getSession();
+
+        if (!$session->has($this->getSessionCartKey($cart))) {
+            $session->set(
+                sprintf('%s.%s', $this->getSessionCartKey($cart), $cart->getChannel()->getCode()),
```

Comment:
ssshhhiiiieeett.... I have coupled it to the channel bundle. I have to reorganize this...